### PR TITLE
More Altivec intrinsics

### PR DESCRIPTION
**Pull request by lu-zero**
Beside the usual json + generated files, I added two additional modifiers in the generator.

### CODE REVIEW

**Comment by nagisa** [comment]
Is there any documentation online which a reviewer could reference when checking the intrinsics?

**Comment by lu-zero** [comment]
Not specifically, that I know.

My way to check has been to compare the llvm output between clang and rustc on simple test-code.

**Comment by lu-zero** [comment]
The C/ELF intrinsics are described [here](http://openpowerfoundation.org/wp-content/uploads/resources/leabi/content/dbdoclet.50655245_pgfId-1138128.html) and historically [http://www.nxp.com/docs/en/reference-manual/ALTIVECPIM.pdf] here, the closest up to date document is [this one](https://openpowerfoundation.org/?resource_lib=power-isa-version-3-0).

File: src/etc/platform-intrinsics/powerpc.json [hunk 1]
```diff
@@ -72,6 +72,55 @@
...
+            "ret": "i(8-32)",
+            "args": ["0", "0"]
+        },
+        {
+            "intrinsic": "subc",
+            "width": [128],
+            "llvm": "vsubcuw",
+            "ret": "u32",
+            "args": ["0", "0"]
+        },
```
Comment (nagisa) [inline src/etc/platform-intrinsics/powerpc.json]: VEC_SUBC [here](http://openpowerfoundation.org/wp-content/uploads/resources/leabi/content/dbdoclet.50655245_56497.html) alludes to both unsigned and signed version existing. Is it intended to only expose unsigned version with the canonical name `subc`?

**Comment by nagisa** [comment]
Everything looks okay, although I wonder if adding just one unsigned variant for `subc` and `addc` was intended.

File: src/etc/platform-intrinsics/powerpc.json (hunk 1 above)
Comment (lu-zero) [inline src/etc/platform-intrinsics/powerpc.json]: The underlying instruction for both signed and unsigned is `vsubcuw`.

I'm currently exposing the altivec intrinsics, then I'll add the VSX and POWER8-only ones.
The safe altivec.h-like-api would live in a separate crate.

Please refer to [altivec.h](https://github.com/llvm-mirror/clang/blob/master/lib/Headers/altivec.h) to see what I'm miming :)

**Comment by lu-zero** [comment]
Yes it was intended.

**Pull request merged**